### ISSUE TITLE
System.TypeInitializationException : The type initializer for 'Grpc.Core.Internal.PlatformApis' threw an exception.

### ISSUE THREAD

**Issue by ronnylt** [grpc/grpc#22069]
### What version of gRPC and what language are you using?
- C# 
- Test assembly with the following packages

```
<?xml version="1.0" encoding="utf-8"?>
<packages>
  <package id="Grpc.Core" version="2.27.0" targetFramework="net472" />
  <package id="Grpc.Core.Api" version="2.27.0" targetFramework="net472" />
  <package id="NUnit" version="3.12.0" targetFramework="net472" />
  <package id="System.Buffers" version="4.5.0" targetFramework="net472" />
  <package id="System.Memory" version="4.5.3" targetFramework="net472" />
  <package id="System.Numerics.Vectors" version="4.5.0" targetFramework="net472" />
  <package id="System.Runtime.CompilerServices.Unsafe" version="4.7.0" targetFramework="net472" />
</packages>
```

### What operating system (Linux, Windows,...) and version?
macOS 10.13.x

### What runtime / compiler are you using (e.g. python version or version of gcc)
.Net Framework v4.7.2

### What did you do?
Created  a Unit Test Project with a simple test like this one

```csharp      
var channel = new Channel("xxx.xxxx.xxxx.xxxx", ChannelCredentials.Insecure);
var client = new Ops.OpsClient(channel);
var reply = client.Status(new StatusRequest());
channel.ShutdownAsync().Wait();
```

### What did you see instead?
```

System.TypeInitializationException : The type initializer for 'Grpc.Core.Internal.PlatformApis' threw an exception.
  ----> System.Reflection.TargetInvocationException : Exception has been thrown by the target of an invocation.
  ----> System.MissingMethodException : Attempted to access a missing method.
  at Grpc.Core.GrpcEnvironment.get_IsNativeShutdownAllowed () [0x00000] in T:\src\github\grpc\src\csharp\Grpc.Core\GrpcEnvironment.cs:434 
  at Grpc.Core.GrpcEnvironment.GrpcNativeInit () [0x00000] in T:\src\github\grpc\src\csharp\Grpc.Core\GrpcEnvironment.cs:364 
  at Grpc.Core.GrpcEnvironment..ctor () [0x0001e] in T:\src\github\grpc\src\csharp\Grpc.Core\GrpcEnvironment.cs:302 
  at Grpc.Core.GrpcEnvironment.AddRef () [0x00028] in T:\src\github\grpc\src\csharp\Grpc.Core\GrpcEnvironment.cs:78 
  at Grpc.Core.Channel..ctor (System.String target, Grpc.Core.ChannelCredentials credentials, System.Collections.Generic.IEnumerable`1[T] options) [0x00041] in T:\src\github\grpc\src\csharp\Grpc.Core\Channel.cs:70 
  at Grpc.Core.Channel..ctor (System.String target, Grpc.Core.ChannelCredentials credentials) [0x00000] in T:\src\github\grpc\src\csharp\Grpc.Core\Channel.cs:55 
  at TestProject1.HelloWorldTest.Greet () [0x00001] in /Users/ronny.lopez/Workspace/tangelo/unity-library/unity-library/TestProject1/Tests.cs:36 
  at TestProject1.Tests.Test1 () [0x00008] in /Users/ronny.lopez/Workspace/tangelo/unity-library/unity-library/TestProject1/Tests.cs:16 
  at (wrapper managed-to-native) System.Reflection.RuntimeMethodInfo.InternalInvoke(System.Reflection.RuntimeMethodInfo,object,object[],System.Exception&)
  at System.Reflection.RuntimeMethodInfo.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x0006a] in /Users/builder/jenkins/workspace/build-package-osx-mono/2019-02/external/bockbuild/builds/mono-x64/mcs/class/corlib/System.Reflection/RuntimeMethodInfo.cs:391 
--TargetInvocationException
  at System.Reflection.RuntimePropertyInfo.GetValue (System.Object obj, System.Object[] index) [0x0006e] in /Users/builder/jenkins/workspace/build-package-osx-mono/2019-02/external/bockbuild/builds/mono-x64/mcs/class/corlib/System.Reflection/RuntimePropertyInfo.cs:390 
  at System.Reflection.PropertyInfo.GetValue (System.Object obj) [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2019-02/external/bockbuild/builds/mono-x64/external/corefx/src/Common/src/CoreLib/System/Reflection/PropertyInfo.cs:41 
  at Grpc.Core.Internal.PlatformApis..cctor () [0x00092] in T:\src\github\grpc\src\csharp\Grpc.Core\Internal\PlatformApis.cs:76 
--MissingMethodException
  at (wrapper managed-to-native) UnityEngine.Application.get_platform()
  at System.Reflection.RuntimePropertyInfo.StaticGetterAdapterFrame[R] (System.Reflection.RuntimePropertyInfo+StaticGetter`1[R] getter, System.Object obj) [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2019-02/external/bockbuild/builds/mono-x64/mcs/class/corlib/System.Reflection/RuntimePropertyInfo.cs:341 
  at System.Reflection.RuntimePropertyInfo.GetValue (System.Object obj, System.Object[] index) [0x0005f] in /Users/builder/jenkins/workspace/build-package-osx-mono/2019-02/external/bockbuild/builds/mono-x64/mcs/class/corlib/System.Reflection/RuntimePropertyInfo.cs:388

```

**Comment by snarlynarwhal** [comment]
I am having the same issue on Windows 10 using .NET Core 3.1 when creating a server:
```
var server = new global::Grpc.Core.Server
{
    Services = { service },
    Ports = { new ServerPort("localhost", 12345, ServerCredentials.Insecure) }
};
```

**Comment by jtattermusch** [comment]
Can you provide a project that reproduces the problem (e.g. in form of a git repo)?

**Comment by snarlynarwhal** [comment]
I ended up getting it working. Unfortunately, I don't recall what caused the issue in the first place.

**Comment by motive-ray** [comment]
I want to chip in as I've been scouring the issues for a similar problem I'm having. I was getting the similar error logs as above, importantly a reference to UnityEngine in a web server project that seemed way out of place. I actually that I did have a reference to UnityEngine.dll that was added by the QRCoder library (see https://github.com/codebude/QRCoder/issues/95). Once removing the reference to both QRCoder and UnityEngine, my problem was resolved.
